fearless/Modules/Staking/StakingPayoutConfirmation/StakingPayoutConfirmationInteractor.swift

```diff
@@ -84,237 +105,295 @@ final class StakingPayoutConfirmationInteractor {
             return
         }
 
-        let options = DataProviderObserverOptions(
-            alwaysNotifyOnRefresh: false,
-            waitsInProgressSyncOnAdd: false
-        )
-        balanceProvider.addObserver(
-            self,
-            deliverOn: .main,
-            executing: updateClosure,
-            failing: failureClosure,
-            options: options
-        )
-    }
+        do {
+            let rewardDestination = try RewardDestination(
+                payee: payee,
+                stashItem: stashItem,
+                chain: chain
+            )
 
-    private func subscribeToPriceChanges() {
-        let updateClosure = { [weak self] (changes: [DataProviderChange<PriceData>]) in
-            if changes.isEmpty {
-                self?.presenter.didReceive(price: nil)
-            } else {
-                for change in changes {
-                    switch change {
-                    case let .insert(item), let .update(item):
-                        self?.presenter.didReceive(price: item)
-                    case .delete:
-                        self?.presenter.didReceive(price: nil)
+            switch rewardDestination {
+            case .restake:
+                presenter.didReceiveRewardDestination(result: .success(.restake))
+
+            case let .payout(payoutAddress):
+                let queryOperation = accountRepository
+                    .fetchOperation(by: payoutAddress, options: RepositoryFetchOptions())
+
+                queryOperation.completionBlock = {
+                    DispatchQueue.main.async {
+                        do {
+                            let account = try queryOperation.extractNoCancellableResultData()
+
+                            let displayAddress = DisplayAddress(
+                                address: payoutAddress,
+                                username: account?.username ?? ""
+                            )
+
+                            let result: RewardDestination = .payout(account: displayAddress)
+
+                            self.presenter.didReceiveRewardDestination(result: .success(result))
+                        } catch {
+                            self.presenter.didReceiveRewardDestination(result: .failure(error))
+                        }
                     }
                 }
+
+                operationManager.enqueue(operations: [queryOperation], in: .transient)
             }
+        } catch {
+            logger?.error("Did receive reward destination error: \(error)")
         }
+    }
 
-        let failureClosure = { [weak self] (error: Error) in
-            self?.presenter.didReceive(priceError: error)
-            return
+    private func createFeeOperationWrapper() -> CompoundOperationWrapper<Decimal>? {
+        guard let batches = batches else { return nil }
+
+        let feeBatches: [Batch] = batches.enumerated().compactMap {
```

Comment:
Why it is so complicated? Please, do something like `let fetchBatches = batches.count > 1 ? [batches[0], batches[count-1]] : [batches[0]]